### DISCUSSION TITLE
Can orama actually work with cloudflare workers?

### DISCUSSION THREAD

**Discussion by djwitcat** [oramasearch/orama#598] [Q&A] [answered]
According to the introduction, orama can run on edge networks. But in my understanding, each workers access runs in a new context, which may mean that data must be re-inserted and indexed every time orama. Even if the data to be searched is very small, assuming there is only 300kb of data, assuming that searching 5 times in 1 second requires re-creating the structure required by orama in the memory 5 times. Is my understanding correct?
I thought kv or durable objects would be helpful before, but after understanding it, I think the data still has to be imported into the workers' memory before each search, so they won't be of much help. I want to know the correct way of orama working in edge environment.

**Comment by micheleriva** [comment, marked as answer]
Hi @group900-3, you can try Orama running on Cloudflare Workers on our homepage (https://oramasearch.com). This is a 10MB catalog with 10,000 real-world products; you can easily inspect the network and see where it's running.

Orama has been built specifically to run on browsers and edge networks. The deployment of such a library to a very constrained environment such as an edge network is definitely challenging to achieve, and that's why we offer Orama Cloud - as we set the environment up for you.

You can try it for free with up to 10,000 elements (soon to be 100,000) at https://cloud.oramasearch.com

**Reply by djwitcat** [reply]
I understand now that running on edge means using "@oramacloud/client" with orama cloud.   
Instead of self-hosted an orama server on workers.

**Reply by micheleriva** [reply]
You can definitely host Orama on Cloudflare Workers on your own, it's just gonna be a bit more difficult than using Orama Cloud - since we do the heavy lifting of scaling the index for you
